### PR TITLE
Add basic VS Code extension using textmate syntax

### PR DESCRIPTION
**Pull request by maan2003**

### CODE REVIEW

File: utils/vscode/.vscode/launch.json
Comment (maan2003) [inline utils/vscode/.vscode/launch.json:1]: This was from [vscode extension template](https://github.com/Microsoft/vscode-generator-code).

File: utils/vscode/.vscode/tasks.json
Comment (maan2003) [inline utils/vscode/.vscode/tasks.json:1]: This was from [vscode extension template](https://github.com/Microsoft/vscode-generator-code).

File: utils/vscode/.vscodeignore
Comment (maan2003) [inline utils/vscode/.vscodeignore:1]: This was from [vscode extension template](https://github.com/Microsoft/vscode-generator-code).

File: utils/vscode/Carbon.plist
Comment (maan2003) [inline utils/vscode/Carbon.plist:1]: not sure why is getting highlighted as invalid here. It is working for me locally.

File: utils/vscode/tsconfig.json
Comment (maan2003) [inline utils/vscode/tsconfig.json:1]: This was from [vscode extension template](https://github.com/Microsoft/vscode-generator-code).

File: utils/vscode/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "configuration": "./language-configuration.json"
+      }
+    ],
+    "grammars": [
+      {
+        "language": "carbon",
+        "scopeName": "source.carbon",
+        "path": "Carbon.plist"
+      }
+    ]
```
Comment (maan2003) [inline utils/vscode/package.json:34]: important part

File: utils/vscode/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  "contributes": {
+    "languages": [
+      {
+        "id": "carbon",
+        "aliases": [],
+        "extensions": [
+          ".carbon"
+        ],
+        "configuration": "./language-configuration.json"
+      }
```
Comment (maan2003) [inline utils/vscode/package.json:26]: important part

**Comment by maan2003** [comment]
Updated to not include VSCode template code.

File: utils/vscode/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "carbon-lang",
+  "displayName": "carbon-lang",
+  "description": "Carbon Language support",
+  "version": "0.0.1",
+  "repository": {
+    "url": "https://github.com/carbon-language/carbon-lang"
+  },
+  "engines": {
+    "vscode": "^1.70.1"
```
Comment (maan2003) [inline utils/vscode/package.json:10]: I only tested 1.70.1 and 1.79.2
Likely older versions are also supported but this is old enough (August 2022)

**Review by josh11b — APPROVED** [review]

**Pull request merged**